Modules/_datetimemodule.c

```diff
@@ -5514,19 +5517,27 @@ datetime_utcfromtimestamp(PyObject *cls, PyObject *args)
 static PyObject *
 datetime_strptime(PyObject *cls, PyObject *args)
 {
-    static PyObject *module = NULL;
     PyObject *string, *format;
 
     if (!PyArg_ParseTuple(args, "UU:strptime", &string, &format))
         return NULL;
 
+    PyObject *result = NULL;
+    PyObject *current_mod = NULL;
+    datetime_state *st = GET_CURRENT_STATE(current_mod);
+    PyObject *module = st->strptime;
     if (module == NULL) {
         module = PyImport_ImportModule("_strptime");
-        if (module == NULL)
-            return NULL;
+        if (module == NULL) {
+            goto Done;
```

Comment:
Nit; `goto exit` is more commonly used throughout the code base.

```suggestion
            goto exit;
```